### ISSUE TITLE
Silent renew not working with v3.0.2 

### ISSUE THREAD

**Issue by thecepe** [damienbod/angular-auth-oidc-client#136]
We are successfully using silent renew of the access token with version 1.3.19. 
After upgrading to version 3.0.2 the token renewal is no longer performed. The log shows no errors. 

Is there anything I missed when upgrading to 3.0.x? I left all configuration settings unchanged and this is the config we are using:

`openIDImplicitFlowConfiguration.stsServer = environment.keycloakBaseUrl + '/auth/realms/' + AppConst.OIDC_REALM;
openIDImplicitFlowConfiguration.redirect_url = window.location.origin + platformLocation.getBaseHrefFromDOM() + PAGE_AUTOLOGIN;
openIDImplicitFlowConfiguration.client_id = AppConst.OIDC_CLIENT_ID;
openIDImplicitFlowConfiguration.response_type = 'id_token token';
openIDImplicitFlowConfiguration.scope = 'openid email profile';
openIDImplicitFlowConfiguration.post_logout_redirect_uri = window.location.origin;
openIDImplicitFlowConfiguration.override_well_known_configuration = false;
openIDImplicitFlowConfiguration.auto_userinfo = false;
openIDImplicitFlowConfiguration.log_console_warning_active = true;
openIDImplicitFlowConfiguration.log_console_debug_active = environment.debug;
openIDImplicitFlowConfiguration.max_id_token_iat_offset_allowed_in_seconds = 10;
openIDImplicitFlowConfiguration.silent_renew = true;
openIDImplicitFlowConfiguration.storage = sessionStorage;
openIDImplicitFlowConfiguration.trigger_authorization_result_event = true;
openIDImplicitFlowConfiguration.silent_renew_offset_in_seconds = 10;`

One other strange thing: 
I set _silent_renew_offset_in_seconds_ to a negative value (-10) to refresh the token before it expires. But with this setting the token expires before renewal. If I set an positive value (10) this problem does not occur. The API documentation says that it should be the other way around.

**Comment by damienbod** [comment]
@thecepe So the problem is when I set a negative value for the silent_renew_offset_in_seconds when using silent renew? 

What is the lifespan of the id_token and the access_token? These should be larger than the silent_renew_offset_in_seconds  + the clock skew

6000 for the id_token
6300 for the access_token would be good values

The silent renew would then be triggered every 10 minutes - the silent_renew_offset_in_seconds

Greetings Damien

**Comment by thecepe** [comment]
@damienbod thanks for your response. I think I mixed up two problems in my first message, sorry for that.

The major problem is that with version 3.0.2 the token renewal is no longer performed (for both negative and positive settings of _silent_renew_offset_in_seconds_). 
I set the token lifespan to 10 minutes and double checked the token (  "iat": 1512996503, "exp": 1512997103 -> lifespan: 600). Did I miss something or is your proposed lifespan to long by factor 10 (6000 seconds would be more than 1.5 h) ? 
With the used IDP I'm only able to set a lifetime for both id_token and access_token. So the lifetime is set for both to 10 minutes.
_silent_renew_offset_in_seconds_ is set to 10 and the system clock is set via ntp.
So the token lifetime is longer than silent_renew_offset_in_seconds + the clock skew


With version 1.3.19 silent renew was working. But at first I sometimes got problems with tokens that where just expired and the review wasn't finished yet. So I tried to trigger the renewal a couple of seconds earlier by setting _silent_renew_offset_in_seconds_ to a negative value (-10) as described in the [API documentation for silent_renew_offset_in_seconds](https://github.com/damienbod/angular-auth-oidc-client/blob/master/API_DOCUMENTATION.md#silent_renew_offset_in_seconds). 
That didn't solve that problem but setting a positive value for _silent_renew_offset_in_seconds_ did.
